transport/sessionStateMachine/transitions.ts

```diff
@@ -112,11 +116,13 @@ export const SessionStateGraph = {
       listeners: SessionBackingOffListeners,
     ): SessionBackingOff => {
       const carriedState = inheritSharedSession(oldSession);
+      const graceExpiryTime = oldSession.graceExpiryTime;
```

Comment:
nit: maybe we can do

```ts
function inheritSharedSessionWithGrace(
  session: IdentifiedSessionWithGracePeriod
): IdentifiedSessionWithGracePeriodProps {
  return {
    ...inheritSharedSession(session),
    graceExpiryTime: session.graceExpiryTime
  };
}
```

```suggestion
      const carriedState = inheritSharedSessionWithGrace(oldSession);
```